core/shared/src/main/scala/zio/stm/TArray.scala

```diff
@@ -157,55 +211,72 @@ class TArray[A] private (val array: Array[TRef[A]]) extends AnyVal {
 
-  /** Atomically compute the greatest element in the array, if it exists. */
+  /**
+   * Atomically compute the greatest element in the array, if it exists.
+   */
   final def maxOption(implicit ord: Ordering[A]): STM[Nothing, Option[A]] =
     reduceOption((acc, a) => if (ord.gt(a, acc)) a else acc)
 
-  /** Atomically compute the least element in the array, if it exists. */
+  /**
+   * Atomically compute the least element in the array, if it exists.
+   */
   final def minOption(implicit ord: Ordering[A]): STM[Nothing, Option[A]] =
     reduceOption((acc, a) => if (ord.lt(a, acc)) a else acc)
 
-  /** Atomically reduce the array, if non-empty, by a binary operator. */
+  /**
+   * Atomically reduce the array, if non-empty, by a binary operator.
+   */
   final def reduceOption(op: (A, A) => A): STM[Nothing, Option[A]] =
     if (array.isEmpty) STM.succeed(None)
     else
       array.head.get
         .flatMap(h => new TArray(array.tail).fold(h)((acc, a) => op(acc, a)))
         .map(Some(_))
 
-  /** Atomically reduce the array, if non-empty, by an STM binary operator. */
+  /**
+   * Atomically reduce the array, if non-empty, by an STM binary operator.
+   */
   final def reduceOptionM[E](op: (A, A) => STM[E, A]): STM[E, Option[A]] =
-    foldM[E, Option[A]](None)(
-      (optAcc, a) =>
-        optAcc match {
-          case Some(acc) => op(acc, a).map(Some(_))
-          case _         => STM.succeed(Some(a))
-        }
-    )
+    foldM[E, Option[A]](None) { (optAcc, a) =>
+      optAcc match {
+        case Some(acc) => op(acc, a).map(Some(_))
+        case _         => STM.succeed(Some(a))
+      }
+    }
 
-  /** Atomically updates all [[TRef]]s inside this array using pure function. */
+  /**
+   * Atomically updates all elements using pure function.
+   */
   final def transform(f: A => A): STM[Nothing, Unit] =
     array.indices.foldLeft(STM.succeed(())) {
       case (tx, idx) => array(idx).update(f) *> tx
     }
 
-  /** Atomically updates all elements using transactional effect. */
+  /**
+   * Atomically updates all elements using transactional effect.
+   */
   final def transformM[E](f: A => STM[E, A]): STM[E, Unit] =
     array.indices.foldLeft[STM[E, Unit]](STM.succeed(())) {
       case (tx, idx) =>
         val ref = array(idx)
         ref.get.flatMap(f).flatMap(a => ref.set(a)).flatMap(_ => tx)
     }
 
-  /** Updates element in the array with given function. */
+  /**
+   * Updates element in the array with given function.
+   */
   final def update(index: Int, fn: A => A): STM[Nothing, A] =
     if (0 <= index && index < array.length) array(index).update(fn)
     else STM.die(new ArrayIndexOutOfBoundsException(index))
 
-  /** Atomically updates element in the array with given transactional effect. */
+  /**
+   * Atomically updates element in the array with given transactional effect.
+   */
   final def updateM[E](index: Int, fn: A => STM[E, A]): STM[E, A] =
```

Comment:
for other parts of zio we have decided to use M suffix only for function involving the zio datatype.
How about we rename these to updateTransactional updateSTM for consistency?

Reply:
I'd leave it for now. We can deprecate this in the future, assuming there's an agreement about that, although, I find the current API to be more convenient.

Reply:
👍